### DISCUSSION TITLE
Is there a  execution time limit in edge function ?

### DISCUSSION THREAD

**Discussion by belliedmonkey** [supabase/supabase#20682] [Questions] [answered]
I want to write an edge function to interact with ChatGPT to complete a background task, which takes about 30 minutes each time. Is this an appropriate use?

**Comment by GaryAustin1** [comment, marked as answer]
https://supabase.com/docs/guides/functions/debugging#understanding-cpu-soft-and-hard-limits
https://supabase.com/docs/guides/functions/debugging#limitations
And the whole page is probably relevant.
